src/librustdoc/html/static/main.js

```diff
@@ -890,6 +890,22 @@
         return "\u2212"; // "\u2212" is '−' minus sign
     }
 
+    $(".search-input").on("keyup",function() {
+        if ($(this).val().length === 0) {
+            window.history.pushState("", "std - Rust", "?search=");
+            $('#main.content').removeClass('hidden');
+            $('#search.content').addClass('hidden');
+        }
+    });
+
+    $('.search-input').on('search', function () {
```

Comment:
Ah, also, `search` event is webkit (safari, epiphany) specific. It is not standard, nor it is supported widely.